File: backend/app/schemas/emergency_contact.py
from pydantic import BaseModel, EmailStr
from typing import Optional
from datetime import datetime


class EmergencyContactCreate(BaseModel):
    """Schema for creating an emergency contact"""
    name: str
    phone: str
    relationship_type: Optional[str] = None
    email: Optional[EmailStr] = None
    is_primary: Optional[int] = 0


class EmergencyContactUpdate(BaseModel):
    """Schema for updating an emergency contact"""
    name: Optional[str] = None
    phone: Optional[str] = None
    relationship_type: Optional[str] = None
    email: Optional[EmailStr] = None
    is_primary: Optional[int] = None


class EmergencyContactResponse(BaseModel):
    """Schema for emergency contact response"""
    id: int
    user_id: int
    name: str
    phone: str
    relationship_type: Optional[str] = None
    email: Optional[str] = None
    is_primary: int
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class EmergencyContactListResponse(BaseModel):
    """Schema for list of emergency contacts"""
    contacts: list[EmergencyContactResponse]
    total: int

File: mental-health-chatbot/frontend/lib/core/models/emergency_contact.dart
class EmergencyContactModel {
  final int? id;
  final String name;
  final String phone;
  final String? relationshipType;
  final String? email;
  final bool isPrimary;
  final DateTime? createdAt;
  final DateTime? updatedAt;

  EmergencyContactModel({
    this.id,
    required this.name,
    required this.phone,
    this.relationshipType,
    this.email,
    this.isPrimary = false,
    this.createdAt,
    this.updatedAt,
  });

  factory EmergencyContactModel.fromJson(Map<String, dynamic> json) {
    return EmergencyContactModel(
      id: json['id'] as int?,
      name: json['name'] as String,
      phone: json['phone'] as String,
      relationshipType: json['relationship_type'] as String?,
      email: json['email'] as String?,
      isPrimary: (json['is_primary'] as int?) == 1,
      createdAt: json['created_at'] != null
          ? DateTime.parse(json['created_at'] as String)
          : null,
      updatedAt: json['updated_at'] != null
          ? DateTime.parse(json['updated_at'] as String)
          : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      if (id != null) 'id': id,
      'name': name,
      'phone': phone,
      if (relationshipType != null) 'relationship_type': relationshipType,
      if (email != null) 'email': email,
      'is_primary': isPrimary ? 1 : 0,
    };
  }

  EmergencyContactModel copyWith({
    int? id,
    String? name,
    String? phone,
    String? relationshipType,
    String? email,
    bool? isPrimary,
    DateTime? createdAt,
    DateTime? updatedAt,
  }) {
    return EmergencyContactModel(
      id: id ?? this.id,
      name: name ?? this.name,
      phone: phone ?? this.phone,
      relationshipType: relationshipType ?? this.relationshipType,
      email: email ?? this.email,
      isPrimary: isPrimary ?? this.isPrimary,
      createdAt: createdAt ?? this.createdAt,
      updatedAt: updatedAt ?? this.updatedAt,
    );
  }
}
